### ISSUE TITLE
Add user and password attribute on the recipe task for windows.

### ISSUE THREAD

**Issue by rogerfdias** [chef-cookbooks/chef-client#178]
I installed the recipe task.rb from the cookbook chef-client for Windows and I noticed that the "run whether user is logged on or not" is not enabled, it seems when you want this option enabled in the windows_task resource you need add the user and password which this task will run, so is a good idea to add this option on the task.rb and also would be a good idea where you can get the password via data_bag.

**Comment by docwhat** [comment]
As a workaround, I do this:

```
include_recipe 'chef-client::task'

chef_gem 'chef-rewind' do
  compile_time true if Chef::Resource::ChefGem.method_defined?(:compile_time)
end

require 'chef/rewind'
rewind 'windows_task[chef-client]' do
  user 'Administrator'
  password admin_password
end
```

**Comment by jtimberman** [comment]
I'm not clear on what the issue is here. The `windows_task` resource in the `task`  recipe already has attributes for the user and password, and has since commit 8822e1d5 (2013-12-16).

Also, there's no need to use `chef-rewind`:

```ruby
resources('windows_task[chef-client]').tap do |r|
  r.user 'Administrator'
  r.password 'Administrator_password'
end
```

**Comment by docwhat** [comment]
In my case, it is because the ['recipe/task.rb`](https://github.com/opscode-cookbooks/chef-client/blob/master/recipes/task.rb) only allows setting the password via an attribute (*shudder* bad security practice) and I have to set it carefully *before* I `include_recipe('recipe/task.rb')`.

In my case, I used `chef-rewind` so that `role[base]` could setup the task with the defaults and then `role[windows]`, which includes `role[base]` can:

1. Change the administrator password.
2. The force the task password to change.

Otherwise I'm either using defaults for the password (from when I built the VM image) or I get errors because the task cannot be created until the password is set.  Not to mention issues when the password changes.

As for using `.tap`... I didn't know that works, @jtimberman.  Since which version of chef? Should I (or you if you feel comfortable) add a comment to this effect to the [chef-rewind](https://github.com/thommay/chef-rewind) `README.md`?  I totally didn't know about this.
